app/models/profile.rb

```diff
@@ -196,6 +196,14 @@ def deactivate_due_to_gpo_expiration
     )
   end
 
+  def deactivate_due_to_ipp_expiration
```

Comment:
Seeing that this largely follows from the equivalent GPO behavior, would we have a similar need to track the "expired at" value as is done with GPO?

Reply:
I think for parity with other profile features we should get an `ipp_enrollment_expired_at` field added to `Profile`, then use the `status_check_completed_at` to seed those values.

Reply:
(That does not need to block this PR though)